runtime/autoload/provider/clipboard.vim

```diff
@@ -156,7 +156,14 @@ function! s:clipboard.get(reg) abort
   elseif s:selections[a:reg].owner > 0
     return s:selections[a:reg].data
   end
-  return s:try_cmd(s:paste[a:reg])
+
+  let clipboard_data = s:try_cmd(s:paste[a:reg])
+  if get(s:selections[a:reg].data, 0, []) == clipboard_data
+    " When system clipboard return is same as our cache return the cache
+    " As it caontains mode information
```

Comment:
"as it contains"

Reply:
Fixed . @clason did you test it on macOS ?

Reply:
Seems to work on a brief test.

Reply:
> I'd trust a functional test more, though :P

me too